src/main/java/data/TaskManager.java

```diff
@@ -47,21 +55,50 @@ public static void addManager(Scanner scanner, WeekView weekView, TaskManager ta
         System.out.println("Task added.");
     }
 
-    public static void deleteManager(Scanner scanner, WeekView weekView, TaskManager taskManager)
-            throws DateTimeParseException, TaskManagerException {
-
-        System.out.println("Enter the date for the task to delete (dd/MM/yyyy):");
+    public static void updateManager(Scanner scanner, WeekView weekView, TaskManager taskManager) throws TaskManagerException {
+        System.out.println("Enter the date for the task you wish to update (dd/MM/yyyy):");
         LocalDate date = getStringFromUser(scanner);
 
         checkIfDateInCurrentWeek(date, weekView);
 
+        listTasksAtDate(taskManager, date, "Enter the task number of the task you wish to update:");
+
+        int taskNumber;
+        String updatedDescription;
+
+        try {
+            taskNumber = Integer.parseInt(scanner.nextLine().trim());
+
+            System.out.println("Enter the updated task description:");
+            updatedDescription = scanner.nextLine().trim();
+
+            updateTask(date, taskNumber - 1, updatedDescription);
+            System.out.println("Task updated.");
+        } catch (NumberFormatException e) {
+            System.out.println("Invalid task number. Please try again.");
+        }
+
+    }
+
+    private static void listTasksAtDate(TaskManager taskManager, LocalDate date, String message) throws TaskManagerException {
         List<String> dayTasks = taskManager.getTasksForDate(date);
         checkIfDateHasTasks(dayTasks);
 
-        System.out.println("Enter the task number to delete:");
+        System.out.println(message);
         for (int i = 0; i < dayTasks.size(); i++) {
             System.out.println((i + 1) + ". " + dayTasks.get(i));
         }
+    }
+
+    public static void deleteManager(Scanner scanner, WeekView weekView, TaskManager taskManager)
+            throws DateTimeParseException, TaskManagerException {
+
+        System.out.println("Enter the date for the task to delete (dd/MM/yyyy):");
+        LocalDate date = getStringFromUser(scanner);
+
+        checkIfDateInCurrentWeek(date, weekView);
+
+        listTasksAtDate(taskManager, date, "Enter the task number to delete:");
```

Comment:
Good use of SLAP! We could consider moving this print statement to be handled by the UI classes.